module/userProjects/helper.js

```diff
@@ -4076,10 +4080,41 @@ function _updateUserProfileBasedOnUserRoleInfo(userProfile, userRoleInformation)
                 userProfile.userRoleMismatchFoundAndUpdated = true;
             }
 
+
             // Create location only object from userRoleInformation
             let userRoleInformationLocationObject = _.omit(userRoleInformation, ['role']);
-            
+
+////check whether userRoleInformation(reqbody) and (db )data matches
+let matchingDistrict, matchingState, matchingBlock, matchingSchool;
+
+for (const eachLocation of userProfile.userLocations) {
+  if (!matchingDistrict && eachLocation.type === 'district' && eachLocation.id === userRoleInformationLocationObject.district) {
+    matchingDistrict = eachLocation;
+  }
+
+  if (!matchingState && eachLocation.type === 'state' && eachLocation.id === userRoleInformationLocationObject.state) {
+    matchingState = eachLocation;
+  }
+
```

Comment:
make it more dynamic and use objectkeys to get all the keys and then validate @praveenKDass